src/PIL/features.py

```diff
@@ -76,14 +76,13 @@ def get_supported_features():
 
 
 def check(feature):
-    return (
-        feature in modules
-        and check_module(feature)
-        or feature in codecs
-        and check_codec(feature)
-        or feature in features
-        and check_feature(feature)
-    )
+    if feature in modules:
+        return check_module(feature)
+    if feature in codecs:
+        return check_codec(feature)
+    if feature in features:
+        return check_feature(feature)
+    raise ValueError("Unknown feature '%s'." % feature)
```

Comment:
I can see why this could be helpful, but it is backwards incompatible.

Reply:
What do you think about changing this to emit a warning? Then the behavior would be the same, but then library users (internal and external) can see that a programming mistake was made.

Or would you rather I drop the commit?

Reply:
I've dropped the backwards incompatible aspect of this change. We can always revisit or reconsider at a later date and it is not directly necessary for the other benefits presented. 

Thanks for the feedback.

Reply:
Moved this to  #4438 so it can be considered and reviewed in isolation